source/base/mpi.cc

```diff
@@ -1468,6 +1501,91 @@ namespace Utilities
       std::pair<types::global_dof_index, types::global_dof_index>,
       unsigned int>;
 
+
+
+    CollectiveMutex::CollectiveMutex()
+      : locked(false)
+      , request(MPI_REQUEST_NULL)
+    {
+      Utilities::MPI::MPI_InitFinalize::register_request(request);
+    }
+
+
+
+    CollectiveMutex::~CollectiveMutex()
+    {
+      Assert(
+        !locked,
+        ExcMessage(
+          "Error: MPI::CollectiveMutex is still locked while being destroyed!"));
+
+      Utilities::MPI::MPI_InitFinalize::unregister_request(request);
+    }
+
+
+
+    void
+    CollectiveMutex::lock(MPI_Comm comm)
+    {
+      (void)comm;
+
+      Assert(
+        !locked,
+        ExcMessage(
+          "Error: MPI::CollectiveMutex needs to be unlocked before lock()"));
+
+#ifdef DEAL_II_WITH_MPI
+
+      // TODO: For now, we implement this mutex with a blocking barrier
+      // in the lock and unlock. It needs to be tested, if we can move
+      // to a nonblocking barrier (code disabled below).
+
+      const int ierr = MPI_Barrier(comm);
+      AssertThrowMPI(ierr);
+
+#  if false && DEAL_II_MPI_VERSION_GTE(3, 0)
```

Comment:
This works, because `false` is not a symbol the preprocessor recognizes -- and so replaces by `0`. In general, you probably want to write this as
```suggestion
#  if 0 && DEAL_II_MPI_VERSION_GTE(3, 0)
```